modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -848,4 +873,51 @@ public String toString() {
             return id.toLowerCase(Locale.ROOT);
         }
     }
+
+    static class ScrollConsumableHitsResponse {
+        private final ScrollableHitSource.AsyncResponse asyncResponse;
+        private final List<? extends ScrollableHitSource.Hit> hits;
+        private final AtomicInteger consumedOffset = new AtomicInteger(0);
```

Comment:
I think this can just be an int?

Reply:
That offset can be updated from different threads, as `AbstractAsyncBulkByScrollAction#prepareBulkRequest` runs in the `GENERIC` thread pool, after throttling has been applied.